src/components/data-management/DownloadData.jsx

```diff
@@ -0,0 +1,136 @@
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import downloadData from '../../utils/data-management/downloadExperimentData';
+import downloadTypes from '../../utils/data-management/downloadTypes';
+import { exportQCParameters, filterQCParameters } from '../../utils/exportQCParameters';
+import { loadBackendStatus } from '../../redux/actions/backendStatus/index';
+
+const DownloadData = (props) => {
+  const {
+    activeProjectUuid,
+  } = props;
+  const dispatch = useDispatch();
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
+  const experimentSettings = useSelector((state) => state.experimentSettings);
+  const backendStatus = useSelector((state) => state.backendStatus);
+  const samples = useSelector((state) => state.samples);
+  const projects = useSelector((state) => state.projects);
+
+  useEffect(() => {
+    if (activeProject?.experiments?.length) {
+      dispatch(loadBackendStatus(activeProject?.experiments[0]));
+    }
+  }, [activeProject]);
+
+  const pipelineHasRun = (experimentId) => (
+    (backendStatus[experimentId]?.status.pipeline?.status === pipelineStatus.SUCCEEDED)
+    && activeProject?.experiments?.length
+  );
+  const gem2sHasRun = (experimentId) => (
+    (backendStatus[experimentId]?.status?.gem2s?.status === pipelineStatus.SUCCEEDED)
+   && activeProject?.experiments?.length
+  );
+
+  const allSamplesAnalysed = () => {
+    // Returns true only if there is at least one sample in the currently active
+    // project AND all samples in the project have been analysed.
+    const steps = Object.values(_.omit(experimentSettings?.processing, ['meta']));
+
+    return steps.length > 0
+      && activeProject?.samples?.length > 0
+      && activeProject?.samples?.every((s) => steps[0].hasOwnProperty(s));
+  };
+
+  return (
+    <Dropdown
+      overlay={() => (
+        <Menu>
+          <Menu.Item
+            key='download-raw-seurat'
+            disabled={!gem2sHasRun(activeProject?.experiments[0])}
+            onClick={() => {
+              const experimentId = activeProject?.experiments[0];
+              downloadData(experimentId, downloadTypes.RAW_SEURAT_OBJECT);
+            }}
+          >
+            <Tooltip
+              title={
+                gem2sHasRun(activeProject?.experiments[0])
+                  ? 'Samples have been merged'
+                  : 'Launch analysis to merge samples'
+              }
+              placement='left'
+            >
+              Raw Seurat object (.rds)
+            </Tooltip>
+          </Menu.Item>
+          <Menu.Item
+            key='download-processed-seurat'
+            disabled={
+              !pipelineHasRun(activeProject?.experiments[0])
+            }
+            onClick={() => {
+              // Change if we have more than one experiment per project
+              const experimentId = activeProject?.experiments[0];
+              downloadData(experimentId, downloadTypes.PROCESSED_SEURAT_OBJECT);
```

Comment:
this `downloadData` function is defined in `utils`, but not used anywhere in the code apart from in here. In this case, having it in a separate file only complicates the code, because now you have to be calculating `experimentId` for it, also have to search for it and so on. I suggest you move the function `downloadData` to be defined in this file, remove the `experimentId` from its argument and have `const experimentId = activeProject?.experiments[0];` as the first line of the function. I would also rename `downloadData` to `downloadExperimentData`, because it looks like you are downloading data for the experiment id